docs/building-a-form/about-the-schema-and-uischema-objects.md

```diff
@@ -0,0 +1,222 @@
+  }
+}
+```
+
+### Understanding the `uiSchema` object
+
+The `uiSchema` object was introduced by [react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form#react-jsonschema-form), or *rjsf*, as a means of describing how a form page should be rendered from a `schema`. To generate a form, react-jsonschema-form steps through the schema depth and renders different React components based on the type of data each property in the schema represents. In the US Forms System library, `uiSchema` follows the format described in the [react-jsonschema-form documentation](https://github.com/mozilla-services/react-jsonschema-form#the-uischema-object), with some custom us-forms-system additions. The `schema` and `uiSchema` objects should have a similar structure, with the same fields organized in the same way with these exceptions:
+
+- `uiSchema` doesn't need to contain all the fields found in the `schema` object.
+- `uiSchema` doesn't need a `properties` object for sub-fields.
+
+For example, given this schema:
+
+```js
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    }
+  }
+}
+```
+
+The matching `uiSchema` would be:
+
+```js
+{
+  'ui:title': 'My form',
+  field1: {
+    'ui:title': 'My field'
+  }
+}
+```
+
+For array fields, you must specify an `items` object that contains the fields for each row in the array in the `uiSchema` object:
+
+```js
+{
+  'ui:title': 'My form',
+  toursOfDuty: {
+    items: {
+      branchName: {
+        'ui:title': 'Branch'
+      }
+    }
+  }
+}
+```
+
+##### Configuring `uiSchema` using rjsf options
+
+If you're not already familiar with the rjsf uiSchema options, see the [library docs](https://github.com/mozilla-services/react-jsonschema-form#the-uischema-object). Some commonly used options include:
+
+- [ui:order](https://github.com/mozilla-services/react-jsonschema-form#object-fields-ordering): An array of field names in the order in which they should appear.
+- [ui:widget](https://github.com/mozilla-services/react-jsonschema-form#alternative-widgets): The name of an alternative widget to use for the field, for example, a custom widget called `yesNo`.
+- [ui:field](https://github.com/mozilla-services/react-jsonschema-form#custom-field-components): The name of a custom field.
+- [classNames](https://github.com/mozilla-services/react-jsonschema-form#custom-css-class-names): The class names to put on the component.
+
+##### Configuring `uiSchema` using US Form System options
+
+The us-forms-system code includes additional `uiSchema` functionality not found in the rjsf library.
+
+```js
+{
+  // Used instead of the `title` property in the JSON Schema.
+  'ui:title': '',
+  // It can also be a component, which passes the current form data as a property.
+  'ui:title': ({ formData }) => <legend>{`A ${formData.thing} title`}</legend>,
+
+  // Used instead of the `description` property in the JSON Schema. This can be a string or a React component, and is normally used on
```

Comment:
These comments in the js code block are still a bit long so they require horizontal scrolling. Since the gesture for horizontal scrolling is the same as the one for forward-back (two-finger swipe) I find myself accidentally navigating pages because of this. I would make the comments no longer than the longest code lines so at least they'll need scrolling at the same time.